staging/src/k8s.io/client-go/transport/cache.go

```diff
@@ -138,17 +145,27 @@ func tlsConfigKey(c *Config) (tlsCacheKey, bool, error) {
 		return tlsCacheKey{}, false, err
 	}
 
-	if c.TLS.GetCert != nil || c.Dial != nil || c.Proxy != nil {
+	if c.Proxy != nil {
 		// cannot determine equality for functions
 		return tlsCacheKey{}, false, nil
 	}
+	if c.Dial != nil && (c.DialHolder == nil || c.DialHolder.Dial == nil || reflect.ValueOf(c.DialHolder.Dial).Pointer() != reflect.ValueOf(c.Dial).Pointer()) {
```

Comment:
I think we should verify the address matches in `isValidHolders` and treat a mismatch as an error, instead of silently ignoring/making uncacheable. Same thing for GetCert address comparison below

That simplifies this to:
`if c.Dial != nil && c.DialHolder == nil { return ... }`